services/QuillLMS/client/app/bundles/Grammar/components/conceptsFeedback/conceptFeedback.tsx

```diff
@@ -44,6 +47,16 @@ class ConceptFeedbackComponent extends React.Component<ConceptFeedbackComponentP
     }
   }
 
+  toggleTranslation() {
+    const translated = !this.state.translated
+    this.setState({...this.state, ...{translated: translated}})
+    if (translated) {
```

Comment:
I think this is not the react way, but wanted to get something working and figured we could fix it in post :)

Reply:
so I think all we are worried about is toggling the text change on button click, right? If so, the following would be preferable:

```
toggleTranslation() {
  const { translated } = this.state
  this.setState({ translated: !translated })
}

renderTranslationButton(translatedDescription) {
  const { translated } = this.state
  if(translatedDescription) {
    const buttonText = translated ? "Hide translation" : "Show translation"
    return <button className="button is-info" id='toggle-translation' onClick={this.toggleTranslation}>{buttonText}</button>
  }
}

return {
  const { translatedDescription } = data[conceptFeedbackID]
  ...
  <button className="button is-info" onClick={this.toggleEdit}>Edit Feedback</button>
  <button className="button is-danger" onClick={this.deleteConceptsFeedback}>Delete Concept Feedback</button>
  {renderTranslationButton(translatedDescription)}
}
```

I know a lot of these older files are a bit messy, but we try to use desctructuring whenever possible, especially for props and state.

Reply:
I agree with Eric's proposal here for this and for the corresponding file in Connect. One note: for the `toggleTranslation` function, you can actually just do:

``` 
toggleTranslation() {
  this.setState(prevState => { translated: !prevState.translated }) }
```